R/recipe_run.R

```diff
@@ -530,30 +568,43 @@ orderly_prepare_data <- function(config, info, parameters, envir) {
   } else {
     hash_global <- NULL
   }
+  ret <- list(hash_resources = hash_resources, hash_global = hash_global)
 
-  if (length(missing_packages) > 0) {
-    stop(paste("Missing packages:", 
-               paste(squote(missing_packages), collapse = ", ")))
-  }
+  ret
+}
 
-  ret <- list(data = ldata, hash_resources = hash_resources,
-              hash_global = hash_global, n_dev = n_dev)
+compare_resource_hashes <- function(pre_run_hashes, post_run_hashes) {
+  ## compare resource hashes here
+  found <- names(post_run_hashes)
+  expected <- names(pre_run_hashes)
 
-  if (!is.null(info$connection)) {
-    ret$con <- data[[info$connection]]
+  ## we expected a resource but can't find it.
+  not_found <- which(!(expected %in% found))
+  if (any(not_found)) {
+    stop("Script either did not copy or deleted resources: ",
+         paste(expected[not_found], collapse = ", "))
   }
 
-  for (p in info$packages) {
-    library(p, character.only = TRUE)
-  }
-  for (s in info$sources) {
-    source(s, envir)
+  ## I'm not sure this can ever happen
+  ## Orderly has found a copied resource that was not expected...
+  not_expected <- which(!(found %in% expected))
+  if (any(not_expected)) {
```

Comment:
I don't understand this - would this not throw on any working files that are created - which we already have a system in place for checking for?  (See `unexpected` within this file)

Reply:
codecov says you never managed to trigger this either - I'm pretty sure we can remove this

Reply:
I was working through the all the possible permutations of finding / not finding resources. This is  caught by unexpected.